src/plugins/saved_objects_management/public/lib/resolve_saved_objects.ts

```diff
@@ -84,50 +84,53 @@ async function importIndexPattern(
     type,
     typeMeta,
   } = doc._source;
-  const importedIndexPattern = {
+  const indexPatternSpec: IndexPatternSpec = {
     id: doc._id,
     title,
     timeFieldName,
-  } as IIndexPattern;
+  };
+  let emptyPattern: IndexPattern;
   if (type) {
-    importedIndexPattern.type = type;
+    indexPatternSpec.type = type;
   }
-  addJsonFieldToIndexPattern(importedIndexPattern, fields, 'fields', title);
-  addJsonFieldToIndexPattern(importedIndexPattern, fieldFormatMap, 'fieldFormatMap', title);
-  addJsonFieldToIndexPattern(importedIndexPattern, sourceFilters, 'sourceFilters', title);
-  addJsonFieldToIndexPattern(importedIndexPattern, typeMeta, 'typeMeta', title);
-  Object.assign(emptyPattern, importedIndexPattern);
-
-  let newId = await emptyPattern.create(overwriteAll);
-  if (!newId) {
-    // We can override and we want to prompt for confirmation
-    const isConfirmed = await openConfirm(
-      i18n.translate('savedObjectsManagement.indexPattern.confirmOverwriteLabel', {
-        values: { title },
-        defaultMessage: "Are you sure you want to overwrite '{title}'?",
-      }),
-      {
-        title: i18n.translate('savedObjectsManagement.indexPattern.confirmOverwriteTitle', {
-          defaultMessage: 'Overwrite {type}?',
-          values: { type },
+  addJsonFieldToIndexPattern(indexPatternSpec, fields, 'fields', title);
+  addJsonFieldToIndexPattern(indexPatternSpec, fieldFormatMap, 'fieldFormatMap', title);
+  addJsonFieldToIndexPattern(indexPatternSpec, sourceFilters, 'sourceFilters', title);
+  addJsonFieldToIndexPattern(indexPatternSpec, typeMeta, 'typeMeta', title);
+  try {
+    emptyPattern = await indexPatterns.createAndSave(indexPatternSpec, overwriteAll, true);
+  } catch (err) {
+    if (err instanceof DuplicateIndexPatternError) {
+      // We can override and we want to prompt for confirmation
+      const isConfirmed = await openConfirm(
+        i18n.translate('savedObjectsManagement.indexPattern.confirmOverwriteLabel', {
+          values: { title },
+          defaultMessage: "Are you sure you want to overwrite '{title}'?",
         }),
-        confirmButtonText: i18n.translate(
-          'savedObjectsManagement.indexPattern.confirmOverwriteButton',
-          {
-            defaultMessage: 'Overwrite',
-          }
-        ),
-      }
-    );
+        {
+          title: i18n.translate('savedObjectsManagement.indexPattern.confirmOverwriteTitle', {
+            defaultMessage: 'Overwrite {type}?',
+            values: { type },
+          }),
+          confirmButtonText: i18n.translate(
+            'savedObjectsManagement.indexPattern.confirmOverwriteButton',
+            {
+              defaultMessage: 'Overwrite',
+            }
+          ),
+        }
+      );
 
-    if (isConfirmed) {
-      newId = (await emptyPattern.create(true)) as string;
-    } else {
-      return;
+      if (isConfirmed) {
+        emptyPattern = await indexPatterns.createAndSave(indexPatternSpec, true, true);
```

Comment:
So we assume it couldn't throw here?

Reply:
In this case we're overriding any existing index patterns so `DuplicateIndexPatternError` won't throw.

As best I can tell the handling of errors (or lack thereof) has been maintained.